symfony/plugins/orangehrmAttendancePlugin/Api/EmployeeAttendanceRecordAPI.php

```diff
@@ -0,0 +1,350 @@
+    public const PARAMETER_TIME = 'time';
+    public const PARAMETER_TIMEZONE_OFFSET = 'timezoneOffset';
+    public const PARAMETER_NOTE = 'note';
+
+    private ?bool $canUserChangeCurrentTime = null;
+    private ?bool $canUserModifyAttendance = null;
+    private ?bool $canSupervisorModifyAttendance = null;
+
+    /**
+     * @return bool
+     */
+    public function canUserChangeCurrentTime(): bool
+    {
+        if (is_null($this->canUserChangeCurrentTime)) {
+            $this->canUserChangeCurrentTime = $this->getAttendanceService()->canUserChangeCurrentTimeConfiguration();
+        }
+        return $this->canUserChangeCurrentTime;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getAll(): EndpointResult
+    {
+        throw $this->getNotImplementedException();
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getValidationRuleForGetAll(): ParamRuleCollection
+    {
+        throw $this->getNotImplementedException();
+    }
+
+    /**
+     * @inheritDoc
+     * @throws Exception
+     */
+    public function create(): EndpointResourceResult
+    {
+        try {
+            $attendanceRecord = new AttendanceRecord();
+            //get request params and convert the keys of the array into variables
+            extract($this->getCommonRequestParams());
+            $attendanceRecord->getDecorator()->setEmployeeByEmpNumber($empNumber);
+            $punchInTimestamp = $this->extractTimestamp($date, $time, $timezoneOffset);
+            $punchInUTCDateTime = $this->getDateTimeHelper()->getUTCDateTime($punchInTimestamp, $timezoneOffset);
+            $overlappingPunchInRecords = $this->getAttendanceService()
+                ->getAttendanceDao()
+                ->checkForPunchInOverLappingRecords($punchInUTCDateTime, $empNumber);
+            if ($overlappingPunchInRecords) {
+                throw AttendanceServiceException::punchInOverlapFound();
+            }
+            $this->setPunchInAttendanceRecord(
+                $attendanceRecord,
+                AttendanceRecord::STATE_PUNCHED_IN,
+                $punchInUTCDateTime,
+                $this->getDateTimeHelper()->getDateTimeByTimestamp($punchInTimestamp),
+                $timezoneOffset,
+                $note
+            );
+            $attendanceRecord = $this->getAttendanceService()->getAttendanceDao()->savePunchRecord($attendanceRecord);
+            return new EndpointResourceResult(AttendanceRecordModel::class, $attendanceRecord);
+        } catch (AttendanceServiceException $e) {
+            throw $this->getBadRequestException($e->getMessage());
+        }
+    }
+
+    /**
+     * @return array
+     */
+    protected function getCommonRequestParams(): array
+    {
+        $empNumber = $this->getRequestParams()->getInt(
+            RequestParams::PARAM_TYPE_ATTRIBUTE,
+            CommonParams::PARAMETER_EMP_NUMBER,
+            $this->getAuthUser()->getEmpNumber()
+        );
```

Comment:
hope this is fine, since we are getting from attribute. But you can separate this to a method, and override in child class.

Reply:
This is working